packages/atlas/src/hooks/useChannelForm.ts

```diff
@@ -189,6 +184,71 @@ export const useChannelForm = (props: FormType) => {
   )
 
   // set default values for editing channel
+  const handleSubmit = (onCompleted?: (channelId: string) => void) =>
+    createSubmitHandler(async (data) => {
+      if (!joystream || !memberId || !accountId) {
+        return
+      }
+
+      if (!channelBucketsCount && !newChannel) {
+        SentryLogger.error('Channel buckets count is not set', 'CreateEditChannelView')
+        return
+      }
+
+      const metadata: ChannelInputMetadata = {
+        ...(dirtyFields.title ? { title: data.title?.trim() ?? '' } : {}),
+        ...(dirtyFields.description ? { description: data.description?.trim() ?? '' } : {}),
+        ...(dirtyFields.language || newChannel ? { language: data.language } : {}),
+        ...(dirtyFields.isPublic || newChannel ? { isPublic: data.isPublic } : {}),
+        ownerAccount: accountId ?? '',
+      }
+      let channelId = ''
+      await handleChannelSubmit(
+        {
+          metadata,
+          channel,
+          newChannel,
+          assets: {
+            avatarPhoto: data.avatar,
+            coverPhoto: data.cover,
+          },
+          refetchChannel: isEditType(props) ? props.refetchChannel : undefined,
+          fee: newChannel ? createChannelFee : updateChannelFee,
+        },
+        (result) => {
+          reset(getValues())
+          channelId = result.channelId
+        },
+        setValue,
+        () => {
+          if (atlasConfig.features.ypp.googleConsoleClientId && atlasConfig.features.ypp.youtubeSyncApiUrl) {
+            setTimeout(() => setShowConnectToYtDialog(true), 2000)
+          }
+
+          // not sure why, but even tho Orion should be in sync initial refetches return empty state
+          // that's why I've introduced this logic to make sure state is in sync before taking actions
```

Comment:
not sure, but I think we should refetch memberships on line 218? This is where onTxSync is called. Could you double-check if this will work?

Reply:
I did on the first version, but it doesn't make sense what should, since `onTxSync` is called even before `onCompleted` and the problem is the Orion is still returning old number of channels for membership.  I would expect a worse outcome when dispatching request even earlier.